ext/enchant/tests/enchant_broker_set_dict_path.phpt

```diff
@@ -51,13 +51,13 @@ if (is_object($broker)) {
 --EXPECTF--
 OK
 
-Deprecated: Function enchant_broker_set_dict_path() is deprecated in %s
+Deprecated: Function enchant_broker_set_dict_path() is deprecated since 8.1 in %s
```

Comment:
I dislike the lone `8.1` in the message. Would it make sense to prefix it with `PHP `? I think this will come up in userspace as well. E.g. if Symfony emits a Doctrine warning, the user can easily confuse which version this is referring to.

Reply:
I wondered about this myself. @kocsismate was interested in the `$since` value for docs, so I'll let him chime in here.

Reply:
Yeah, I was also wondering about it, even I wanted to write a review comment, but then I changed my mind because I realized that adding the PHP prefix only works for bundled extensions. 3rd party ones cannot refer to PHP versions... But I agree the situation is very unfortunate. :(

Reply:
They can refer to `ext-... x.y.z `. I think it's better than nothing, although I would have preferred not having it at all (it's useless information anyway).

Reply:
Would this be an argument for adding a package property as well?

Reply:
> Would this be an argument for adding a package property as well? 

Either we need a `package` property in addition to `version`, or we have to add the package name to the `version` property just like how Ilija wrote. But in the latter case, it shouldn't be called `version` anymore, but `package` I guess. 🤔 

So if it's important to provide exact information about the package and the version separately then I think both properties are needed. If it's enough to provide them as free form text, then a `package` property is what we need.

For me use-case, both options may work.

Reply:
To ensure clarity and usefulness, the since attribute should contain both package and version information. The examples in this PR that use 'since: number' could be more user-friendly with the inclusion of the package name.

@iluuu1994: This information is actually quite valuable. The link between a symbol and its corresponding package isn't always clear, either for humans or machines. Clear and precise documentation is extremely helpful in practice.

I find it surprising how much we're revisiting concepts here, especially considering the insights from Symfony, which have been honed through years of practical experience. Nonetheless, let's proceed and make the best of it. :shrug:

Reply:
@nicolas-grekas What I was referring to being useless is the `$since` property itself. If the project is following semantic versioning, you know the deprecation was added in this major version (not helpful) and will be removed in the next one (helpful). If it doesn't, then all bets are off anyway. `$toBeRemovedIn` or similar would actually be useful.

Reply:
sure thing, but that needs external knowledge to me resolved (which current version of what package you're using)
while everything is available somewhere is a composer map, making it obvious is very valuable in practice (the package name is the most important, but the version is also a nice reminder of how outdated you are).

Reply:
Oh, I realize that we might be talking about different things. I'm talking about userland packages indeed.
For php itself and for extensions, then yes, I agree that "since" is useless. My recommendation would be to not use it.

Reply:
I think the `$since` property can be useful for even internal symbols: my niche use-case is to auto-generate the warning box on manual pages (like the one at https://www.php.net/manual/en/function.gmstrftime.php).

Besides, the property can come handy when one wants to find the change log/upgrading note of a deprecated symbol, even if it's an internal one. If I immediately know the version it was deprecated in then I don't have to start looking up all the possible change logs... I know this from experience, because I upgraded the PHP version of our large proprietary application multiple times.

Reply:
Note: I've spun out applying the attribute into a separate PR #14750 to not further block the merge of the base implementation.

I'll look through this thread and think about it after the merge is done, possibly continuing the discussion in the new PR.